### PR TITLE
Adding an example deployment yaml

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref #1743 
- Updating examples_test to be able to work with experimental objects.
- Adding an example deployment.yaml.

Assigning to @caesarxuchao to review the examples_test changes.

cc @ghodss

### CODE REVIEW

**Comment by pmorie** [comment]
@ironcladlou

**Comment by ironcladlou** [comment]
LGTM

File: examples/examples_test.go [hunk 1]
```diff
@@ -369,7 +379,11 @@ func TestExampleObjectSchemas(t *testing.T) {
 				}
 				//TODO: Add validate method for &schedulerapi.Policy
 			} else {
-				if err := testapi.Default.Codec().DecodeInto(data, expectedType); err != nil {
+				codec := testapi.Default.Codec()
+				if strings.HasSuffix(path, "examples/experimental") {
```
Comment (caesarxuchao) [inline examples/examples_test.go]: This looks questionable to me. Is it possible an experimental example needs to use a v1 object? I think it's more robust to deduce the group by using the expectedType.
My pending PR has added such a function to testapi, https://github.com/kubernetes/kubernetes/pull/14156/files#diff-59998217138141c72ed55d8a701cd00fR214
Do you want to try that?

Comment (nikhiljindal) [inline examples/examples_test.go]: Done

**Comment by nikhiljindal** [comment]
@caesarxuchao Updated code as per your comment. I copied GetCodecForObject from your PR. PTAL

**Comment by caesarxuchao** [comment]
Thank you. LGTM. I guess if I don't change the function in my PR, then neither of us would need to do a rebase?

**Comment by nikhiljindal** [comment]
Actually there was a slight difference in order of imports. Fixed it now, so that none of us will have to rebase.
Will need your LGTM again.

**Comment by caesarxuchao** [comment]
Thank you @nikhiljindal. LGTM.

**Pull request merged**